File: app/src/main/java/doit/study/droid/data/source/local/entities/QuestionTagJoin.java
package doit.study.droid.data.source.local.entities;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.ForeignKey;
import android.arch.persistence.room.Index;
import android.support.annotation.NonNull;

import static android.arch.persistence.room.ForeignKey.CASCADE;

@Entity(
        tableName="question_tag_join",
        primaryKeys={"questionId", "tagId"},
        foreignKeys={
                @ForeignKey(
                        entity=QuestionEntity.class,
                        parentColumns="id",
                        childColumns="questionId",
                        onDelete=CASCADE),
                @ForeignKey(
                        entity=TagEntity.class,
                        parentColumns="id",
                        childColumns="tagId",
                        onDelete=CASCADE)},
        indices={
                @Index(value="questionId"),
                @Index(value="tagId")
        } )
public class QuestionTagJoin {
    @NonNull
    public final long questionId;
    @NonNull
    public final long tagId;

    public QuestionTagJoin(long questionId, long tagId) {
        this.questionId=questionId;
        this.tagId = tagId;
    }
}

File: app/src/main/java/doit/study/droid/data/local/entity/QuestionTagJoin.kt
package doit.study.droid.data.local.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
        foreignKeys = [
            ForeignKey(
                    entity = Question::class,
                    parentColumns = ["id"],
                    childColumns = ["questionId"]
            ),
            ForeignKey(
                    entity = Tag::class,
                    parentColumns = ["id"],
                    childColumns = ["tagId"]
            )
        ],
        indices = [
            Index("questionId"),
            Index("tagId")
        ]
)
data class QuestionTagJoin(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,
    val questionId: Int,
    val tagId: Int
)
